src/controls/arrow.js

```diff
@@ -27,14 +32,22 @@ export default class Arrow extends Component {
   }
 
   render() {
-    const { prevSlide, nextSlide, direction } = this.props;
+    const { prevSlide, nextSlide, direction, customArrow } = this.props;
+    let arrowComponent = null;
+    let buttonClass = 'carousel-arrow-default';
+
+    if (customArrow) {
+      buttonClass = customArrow.className ? customArrow.className : '';
```

Comment:
If they don't supply a className, should it get no className or `carousel-arrow-default`?  Think `<Carousel customArrow={{ left, right, className }} />` vs Think `<Carousel customArrow={{ left, right }} />` . IMHO if the property doesn't exist it should keep the default. otherwise it should use the value and the consumer can pass an empty string to get rid of the default class.

Reply:
Also you could model all this with just a single property with defaultProps providing the `carousel-arrow-default`

Reply:
we need the `carousel-arrow-default` only when the user has not provided a custom arrow. It has the styles specific to the existing version of the arrow which we don't need if a custom arrow is given.